scipy/stats/tests/test_stats.py

```diff
@@ -929,6 +929,58 @@ def test_index_error(self):
             0.0007835762419683435])
         assert_raises(ValueError, stats.spearmanr, x, y, axis=2)
 
+    def test_alternative(self):
+        # Test alternative parameter
+
+        # Simple test - Based on the above ``test_spearmanr_vs_r``
+        x1 = [1, 2, 3, 4, 5]
+        x2 = [5, 6, 7, 8, 7]
+
+        # strong positive correlation
+        expected = (0.82078268166812329, 0.088587005313543798)
+
+        # correlation > 0 -> large "less" p-value
+        res = stats.spearmanr(x1, x2, alternative="less")
+        assert_approx_equal(res[0], expected[0])
+        assert_approx_equal(res[1], 1 - (expected[1] / 2))
+
+        # correlation > 0 -> small "less" p-value
+        res = stats.spearmanr(x1, x2, alternative="greater")
+        assert_approx_equal(res[0], expected[0])
+        assert_approx_equal(res[1], expected[1] / 2)
```

Comment:
Is it worth converting these to access the result using attributes rather than indexes?

Reply:
Sure.

Reply:
This is how you had it originally, I think. If you'd like to update to separate, go for it. But it should also then have separate variables for the expected p-value and statistic, I think.